service/raft/raft_group0_client.cc

```diff
@@ -372,4 +373,44 @@ db::system_keyspace& raft_group0_client::sys_ks() {
     return _sys_ks;
 }
 
+raft_group0_client::query_result_guard::query_result_guard(utils::UUID query_id, raft_group0_client& client)
+    : _query_id{query_id}, _client{&client} {
+    auto it = _client->_results.emplace(_query_id, std::nullopt);
```

Comment:
`it` should be reserved for iterators, here it's not an iterator but a pair <iterator, bool>
I suggest:
`auto [_, emplaced] = ...`